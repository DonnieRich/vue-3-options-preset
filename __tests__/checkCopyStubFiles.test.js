const { vol } = require("memfs");
jest.mock("fs/promises");

const { copyStubFiles } = require("../src/bin/copyStubFiles");

// import all json config files
const { config } = require('../src/config/config.test');
const { BASE_DIR, BASE_STUBS_DIR } = config.get();

describe(copyStubFiles, () => {
    beforeEach(() => {
        vol.reset();
    });

    it("It should pass if the stub files are copied in the new location", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/App.vue`]: "...",
                [`${BASE_STUBS_DIR}/general.scss`]: "...",
                [`${BASE_STUBS_DIR}/HelloWorld.vue`]: "...",
                [`${BASE_STUBS_DIR}/main.js`]: "...",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "old component"
            },
            "/"
        );

        await copyStubFiles(BASE_STUBS_DIR, BASE_DIR, false);
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("It should pass if the stub files are copied in the new location with  SCSS file for Bootstrap", async () => {

        vol.fromJSON(
            {
                [`${BASE_STUBS_DIR}/App.vue`]: "...",
                [`${BASE_STUBS_DIR}/general-bootstrap.scss`]: "...",
                [`${BASE_STUBS_DIR}/HelloWorld.vue`]: "...",
                [`${BASE_STUBS_DIR}/main.js`]: "...",
                [`${BASE_DIR}/justAFile.js`]: "...",
                [`${BASE_DIR}/components/HelloWorld.vue`]: "old component"
            },
            "/"
        );

        await copyStubFiles(BASE_STUBS_DIR, BASE_DIR, true);
        expect(vol.toJSON()).toMatchSnapshot();
    });
});