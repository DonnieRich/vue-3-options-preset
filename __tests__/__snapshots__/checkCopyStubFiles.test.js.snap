// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copyStubFiles It should pass if the stub files are copied in the new location 1`] = `
{
  "/node_modules/vue-3-options-preset/src/stubs/App.vue": "...",
  "/node_modules/vue-3-options-preset/src/stubs/HelloWorld.vue": "...",
  "/node_modules/vue-3-options-preset/src/stubs/general.scss": "...",
  "/node_modules/vue-3-options-preset/src/stubs/main.js": "...",
  "/src/App.vue": "...",
  "/src/components/HelloWorld.vue": "...",
  "/src/justAFile.js": "...",
  "/src/main.js": "...",
  "/src/styles/general.scss": "...",
}
`;
